### ISSUE TITLE
Doesn't work on NixOS

### ISSUE THREAD

**Issue by doronbehar** [natsukagami/mpd-mpris#5]
Hey,

I'd like to add mpd-mpris to NixOS. I've managed to compile it and it seems to be working when I run the executable. From some reason though, `playerctl -l` reports "No players were found" when `mpd-mpris` is running. Back on Arch Linux, I haven't had this problem.

I have d-feet installed and while `mpd-mpris` is running, it doesn't show any `org.mpris.MediaPlayer2.mpd` bus so I think `playerctl` is not the guilty program. Besides that, I've tested your competent :smirk: [mpDris2](https://github.com/eonpatapon/mpDris2) and it works. Trust me though, I'd very much prefer to use mpd-mpris because I hate Python.

Is there any way I can make `mpd-mpris` be more verbose and print more information in it's log? This may be useful for debugging.

**Comment by natsukagami** [comment]
Interesting. I haven't been checking out this repo (nor using it, I'm no
longer in need for mpris) for a long time, but as far as I can remember,
all the code in mpd-mpris are as verbose as it can (maybe except for the
player-updating loop, but that's out of the scope for this.)

If there is such a weird bug, I'd suspect it to be go-dbus's fault. I'm
happy to have a look again at my repo though! However since I'm not very
knowledgeable on dbus, I don't really know where to look for the bug...

**Comment by doronbehar** [comment]
Hey, thanks for getting back to me. I've checked out the go-dbus repo and I can confirm that their examples work fine and I can see it's evidence in d-feet. For instance, there's [`prop.go`](https://github.com/godbus/dbus/blob/master/_examples/prop.go) which seems to be doing something similar to mpd-mpris and it can be seen in d-feet when I run it.

I've been trying pretty hard to debug this but I couldn't find an interesting clue. I have a suspicion that it's related to default dbus configuration files on NixOS. One very suspicious issue is the fact that there's a new `dbus-daemon --syslog --fork --session` process that spawns every time I test `mpd-mpris`, it doesn't even terminate when I terminate the `mpd-mpris` process.

Anyway, these are my pointers so far.. I'll keep on debugging this through system configuration changes.

 BTW, just out of curiosity, why did you stop to have a need for mpris for MPD? I mean, what was your initial use case and how come now you don't need it anymore?

**Comment by natsukagami** [comment]
Thanks for the efforts! I've never used NixOS so afraid I won't be helpful.

For the second question, I was writing this because I was using mpd along with spotify, so I wanted a one-for-all control bar that could automatically (and manually) be switched back and forth between the two.
I don't use spotify anymore (ugh, it never finds songs I like - mysterious), so I'm back with the simple mpd controls (which there are a billion of them) now.

**Comment by doronbehar** [comment]
I've found the issue: If `$DBUS_SESSION_BUS_ADDRESS` environmental variable is not set, mpd-mpris (via it's go-dbus usage) starts it's own `dbus-launch` process and in turn this results in more and more `dbus-daemon` processes just as I've experienced. The main source for this knowledge is from [`dbus-launch(1)`](https://dbus.freedesktop.org/doc/dbus-launch.1.html).

As explained in this man page, I've fixed the issue on my setup by exporting this environmental variable in my `.xinitrc`.

Naturally, this is strictly a NixOS specific issue and I'm sorry if it bothered you. Thanks for attending this never the less.
